streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableTransformValues.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.kstream.internals;
+
+import org.apache.kafka.streams.kstream.ValueTransformerWithKey;
+import org.apache.kafka.streams.kstream.ValueTransformerWithKeySupplier;
+import org.apache.kafka.streams.processor.AbstractProcessor;
+import org.apache.kafka.streams.processor.Processor;
+import org.apache.kafka.streams.processor.ProcessorContext;
+import org.apache.kafka.streams.processor.internals.ForwardingDisabledProcessorContext;
+import org.apache.kafka.streams.state.KeyValueStore;
+
+import java.util.Objects;
+
+
+class KTableTransformValues<K, V, V1> implements KTableProcessorSupplier<K, V, V1> {
+
+    private final KTableImpl<K, ?, V> parent;
+    private final ValueTransformerWithKeySupplier<? super K, ? super V, ? extends V1> transformerSupplier;
+    private final String queryableName;
+    private boolean sendOldValues = false;
+
+    KTableTransformValues(final KTableImpl<K, ?, V> parent,
```

Comment:
For `KTableTransformValuesProcessor` that extends `AbstractProcessor`, we should override its `close()` call as:

```
       @Override
        public void close() {
            valueTransformer.close();
        }
```

Similarly we should close the `valueTransformer` in `Getter` as well. Unfortunately we do not have a `close()` in `KTableValueGetter` yet. If there is no better way to call `valueTransformer.close()` when the topology is closing we should then add this function and make sure it is called when the dependent processor node is being closed.

Reply:
Can't see a better way, so adding `close()` to KTableValueGetter and wiring that up.